src/command/render/output-typst.ts

```diff
@@ -94,12 +98,29 @@ export function typstPdfOutputRecipe(
     : normalizeOutputPath(input, join(inputDir, inputStem + ".pdf"));
 
   // return recipe
-  return {
+  const recipe: OutputRecipe = {
     output,
     keepYaml: false,
     args,
     format: { ...format, pandoc },
     complete,
     finalOutput: pdfOutput ? relative(inputDir, pdfOutput) : undefined,
   };
+
+  // if we have some variant declared, resolve it
+  // (use for opt-out citations extension)
+  if (format.render?.[kVariant]) {
+    const to = format.pandoc.to;
+    const variant = format.render[kVariant];
+
+    recipe.format = {
+      ...recipe.format,
+      pandoc: {
+        ...recipe.format.pandoc,
+        to: `${to}${variant}`,
+      },
+    };
+  }
```

Comment:
We also need to account for the variants so that `-citations` can be passed when rendering to PDF. This is same code as in `output.ts`